src/librustc_resolve/lib.rs

```diff
@@ -3373,7 +3373,11 @@ impl<'a> Resolver<'a> {
         };
 
         let mut err = match (old_binding.is_extern_crate(), binding.is_extern_crate()) {
-            (true, true) => struct_span_err!(self.session, span, E0259, "{}", msg),
+            (true, true) => {
+                let mut e = struct_span_err!(self.session, span, E0259, "{}", msg);
```

Comment:
@GuillaumeGomez this is what I initial tried. The problem is that span_label returns a &mut of the expected match type. So unless we modified all the arms in the match to the oneliner so that it would match to the &mut TYPE, there would be a compiler error for the arm that used the oneliner. Seeing as almost all the arms in the bracket are switching to use the span_label, it shouldn't be a big problem to modify the entire match block.